app-shell/src/buildroot/index.ts

```diff
@@ -88,14 +88,14 @@ export function registerBuildrootUpdate(dispatch: Dispatch): Dispatch {
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         uploadSystemFile(host, path, file)
           .then(() => ({
-            type: 'buildroot:FILE_UPLOAD_DONE',
+            type: 'buildroot:FILE_UPLOAD_DONE' as 'buildroot:FILE_UPLOAD_DONE',
```

Comment:
my bad for not thinking of this earlier, but these `as` casts can all be changed to `as const` instead which is more accurate and safer

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions
```suggestion
            type: 'buildroot:FILE_UPLOAD_DONE' as const,
```